medallion/backends/base.py

```diff
@@ -1,6 +1,25 @@
 
 class Backend(object):
 
+    __SUBCLASS_MAP = dict()
+
+    def __init_subclass__(cls, **kwargs):
```

Comment:
This is a Python 3.6 feature. It'd be pretty easy to do it with a metaclass instead, but I thought I would use the new hotness first to prompt a discussion wrt appetite for dropping 3.5 support.

Reply:
Hi @maybe-sybr, would the change to perform this with a metaclass be a big change to the current work you have made? Recently among the team we discussed dropping support to 2.7 and 3.4 across all the open repos, but considered to keep 3.5 since it is fairly recent. I think in this occasion there is an alternative option as opposed to dropping support right away.

Reply:
It's certainly something I can do, not too much work at all. I'll look into getting that done for you some time today or tomorrow when I have the time :)

Reply:
@emmanvg - this is done now. Check out the updated changeset when you have a chance :)